packages/ui/src/composables/useTrapFocus.ts

```diff
@@ -0,0 +1,92 @@
+import { nextTick, Ref } from 'vue'
+import { useDocument } from './useDocument'
+import { useWindow } from './useWindow'
+
+const TAB_KEYCODE = 9
+const FOCUSABLE_ELEMENTS_SELECTOR = ':where(a, button, input, textarea, select):not([disabled]), *[tabindex]'
+let trapInEl: HTMLElement | null = null
+
+export const useTrapFocus = () => {
+  const document = useDocument()
+  const window = useWindow()
+
+  let focusableElements: HTMLElement[] = []
+  let firstFocusableElement: HTMLElement | null = null
+  let lastFocusableElement: HTMLElement | null = null
+  let isFocusTrapped = false
+
+  const focusFirstElement = () => {
+    if (firstFocusableElement) {
+      firstFocusableElement.focus()
+    }
+  }
+  const focusLastElement = () => {
+    if (lastFocusableElement) {
+      lastFocusableElement.focus()
+    }
+  }
+
+  const onKeydown = (evt: KeyboardEvent) => {
+    const isTabPressed = evt.key === String(TAB_KEYCODE) || evt.keyCode === TAB_KEYCODE
```

Comment:
We can use `KeyboardEvent.code` since we don't need IE support.

```suggestion
    const isTabPressed = evt.code === 'Tab'
```